daml-lf/kv-support/src/test/scala/com/daml/lf/kv/transactions/TransactionTraversalSpec.scala

```diff
@@ -86,19 +87,36 @@ class TransactionTraversalSpec extends AnyFunSuite with Matchers {
     } shouldBe Right(())
   }
 
-  test("traverseTransactionWithWitnesses - transaction parsing error") {
+  test("extractPerPackageWitnesses - extract package witness mapping as expected") {
+    val result = TransactionTraversal.extractPerPackageWitnesses(rawTx)
+    result shouldBe
+      Right(
+        Map(
+          "template_exercise" -> Set("Alice", "Charlie"),
+          "interface_exercise" -> Set("Alice", "Charlie"),
+          "template_create" -> Set("Alice", "Charlie", "Bob"),
+          "template_fetch" -> Set("Alice", "Bob"),
+        )
+      )
+  }
+
+  test("traversing transaction parsing error") {
```

Comment:
Nit: to be consistent with the naming convention used in this test suite, I'd suggest:
```suggestion
  test("traversal - transaction parsing error") {
```